qiskit_machine_learning/algorithms/classifiers/pegasos_qsvc.py

```diff
@@ -0,0 +1,223 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Quantum Pegasos Support Vector Classifier"""
+
+from typing import Optional, Dict, Tuple
+from datetime import datetime
+import numpy as np
+from sklearn.svm import SVC
+from qiskit_machine_learning.kernels.quantum_kernel import QuantumKernel
+from qiskit_machine_learning.exceptions import QiskitMachineLearningError
+
+
+class PegasosQSVC(SVC):
+    r"""Quantum Pegasos Support Vector Classifier
+    This class implements the algorithm developed in [1] and includes some of the methods like
+    ``fit`` and ``predict`` like in QSVC.
+
+    **Example**
+
+    .. code-block::
+        quantum_kernel = QuantumKernel()
+
+        pegasos_qsvc = PegasosQSVC(quantum_kernel=quantum_kernel)
+        pegasos_qsvc.fit(sample_train, label_train)
+        pegasos_qsvc.predict(sample_test)
+
+    **References**
+        [1]: Shalev-Shwartz et al., Pegasos: Primal Estimated sub-GrAdient SOlver for SVM.
+            `Pegasos for SVM <https://home.ttic.edu/~nati/Publications/PegasosMPB.pdf>`_
+
+    """
+
+    def __init__(
+        self,
+        quantum_kernel: Optional[QuantumKernel] = None,
+        C: float = 1,
+        num_steps: int = 1000,
+        verbose: bool = False,
+    ) -> None:
+        """
+        Args:
+            quantum_kernel: QuantumKernel to be used for classification.
+            C: positive regularization parameter
+            num_steps: number of steps in the Pegasos algorithm
+            verbose: whether the time taken for train and predict is printed
+        """
+        self._quantum_kernel = quantum_kernel if quantum_kernel is not None else QuantumKernel()
+
+        super().__init__(C=C)
+        self._num_steps = num_steps
+        self._verbose = verbose
+
+        # these are the parameters being fit and are needed for prediction
+        self._fit_status = False
+        self._alphas: Optional[Dict[int, int]] = None
+        self._x_train: Optional[np.ndarray] = None
+        self._y_train: Optional[np.ndarray] = None
+        self._label_dict: Optional[Dict[int, int]] = {}
```

Comment:
Could you rename `_label_dict` to something without the `_dict` suffix?

Reply:
See 5b8382e35450af16c8b6a766a9be2becd849e947